installer.sh

```diff
@@ -0,0 +1,413 @@
+        fi
+    fi
+}
+
+function call_user() {
+    sh -c "$1" 2>&1 | indent
+}
+
+function check_shasum() {
+    if command -v sha1sum > /dev/null 2>&1; then
+        cmd_shasum="sha1sum"
+        output "  [*] sha1sum is installed" "success"
+    elif command -v shasum > /dev/null 2>&1; then
+        cmd_shasum="shasum -a 1"
+        output "  [*] shasum is installed" "success"
+    else
+        output "  [ ] No sha1sum or shasum available to verify binary" "error"
+        exit_with_error
+    fi
+}
+
+function check_directories() {
+    if [ ! -z $PLATFORMSH_CLI_INSTALL_DIR ]; then
+        dir_bin="${PLATFORMSH_CLI_INSTALL_DIR}"
+    fi
+
+    if ! echo $PATH | grep ${dir_bin} > /dev/null; then
+        binary="${dir_bin}/platform"
+
+        output "  [ ] ${dir_bin} is not in \$PATH.\n" "warning"
+        add_footer_note "  ⚠ The directory \"${dir_bin}\" is not in \$PATH."
+        if echo $SHELL | grep '/bin/zsh' > /dev/null
+        then
+            add_footer_note \
+                "    Run this command to add the directory to your PATH" \
+                "    echo 'export PATH=\"${dir_bin}:\$PATH\"' >> \$HOME/.zshrc"
+        elif echo $SHELL | grep '/bin/bash' > /dev/null
+        then
+            add_footer_note \
+                "    Run this command to add the directory to your PATH" \
+                "    echo 'export PATH=\"${dir_bin}:\$PATH\"' >> \$HOME/.bashrc"
+        else
+            add_footer_note \
+                "    You can add it to your PATH by adding this line at the end of your shell configuration file" \
+                "    export PATH=\"${dir_bin}:\$PATH\""
+        fi
+    else
+        output "  [*] ${dir_bin} is in \$PATH." "success"
+    fi
+
+    output "\nTarget directories" "heading"
+    output "  - Binary will be installed in ${dir_bin}"
+}
+
+function install_platformsh_homebrew() {
+    output "\nInstalling Platform.sh brew tap" "heading"
+    if ! call_user "brew tap $PLATFORMSH_CLI_BREW_TAP"; then
+        output "  could not add tap to brew" "error"
+        exit_with_error
+    fi
+
+    output "\nInstalling Platform.sh CLI formula" "heading"
+
+    # running on Rosetta2?
+    arch=''
+    if [ "$(uname -m)" = "x86_64" ]; then
+        if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]; then
+            arch="arch -arm64"
+        fi
+    fi
+
+    if ! call_user "$arch brew install $PLATFORMSH_CLI_BREW_FORMULA"; then
+        output "  could not install Formula" "error"
+        exit_with_error
+    fi
+}
+
+function install_platformsh_raw() {
+    # Start downloading the right version
+    output "\nDownloading the Platform.sh CLI Tool" "heading"
+
+    output "  Finding the right version for platform ${kernel}/${machine}";
+
+    url="${PLATFORMSH_CLI_URL}/${version}/platform_${version}_${kernel}_${machine}.tar.gz"
+    output "  Downloading ${url}";
+    tmp_dir=$(mktemp -d)
+    tmp_name="platformsh-"$(date +"%s")
+    if ! $cmd_download $url > "${tmp_dir}/${tmp_name}.tgz"; then
```

Comment:
The legacy installer uses auth with GitHub so that installs on app containers do not hit a rate limit:

https://github.com/platformsh/legacy-cli/blob/a224164cb5a74ea84371cce4903059775b33edab/dist/installer.php#L785-L809

This may not be necessary for now, but it would be if this was the recommended and widely used install method

Reply:
https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting

> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the person making requests.

Reply:
Good one, fixed if the `GITHUB_TOKEN` environment variable exists, which fixes GitHub actions and also gives a way out if this fails for a user.